### PR TITLE
Reduce irrelevant build warnings

### LINKED ISSUE #1961: MSVC build warnings
See below output messages. I'll let you work out if any are relevant or need correction

| Description | File name | Line |Column | Project|
|-------------|-----------|--------|-------|---------|
| Tag 'TCL_SUBST' at line 229 of file 'C:/projects/corsixth/DoxyGen/animview.doxygen' has become obsolete. | CUSTOMBUILD | | | doc_animview |
| Tag 'TCL_SUBST' at line 229 of file 'C:/projects/corsixth/DoxyGen/leveledit.doxygen' has become obsolete. | CUSTOMBUILD | | | doc_leveledit |
| 'fopen': This function or variable may be unsafe. Consider using fopen_s instead. To disable deprecation, use _CRT_SECURE_NO_WARNINGS. See online help for details. | CorsixTH\Src\iso_fs.cpp | 374 | 18 | CorsixTH_lib |
| Tag 'COLS_IN_ALPHA_INDEX' at line 986 of file 'C:/projects/corsixth/DoxyGen/animview.doxygen' has become obsolete. | CUSTOMBUILD | | | doc_animview |
| Tag 'TCL_SUBST' at line 229 of file 'C:/projects/corsixth/DoxyGen/corsixth_engine.doxygen' has become obsolete. | CUSTOMBUILD | | | doc_corsixth_engine|
| Tag 'COLS_IN_ALPHA_INDEX' at line 986 of file 'C:/projects/corsixth/DoxyGen/corsixth_engine.doxygen' has become obsolete. | CUSTOMBUILD | | | doc_corsixth_engine |
Tag 'COLS_IN_ALPHA_INDEX' at line 986 of file 'C:/projects/corsixth/DoxyGen/leveledit.doxygen' has become obsolete. | CUSTOMBUILD | | | doc_leveledit |

### PR DESCRIPTION
**Pull request by tobylane**
*Fixes #1961*

**Describe what the proposed change does**
- Remove the old parts of doxygen configureations that it warns about
- Suppress the invalid warning in iso_fs

### CODE REVIEW

File: CorsixTH/Src/iso_fs.cpp [hunk 1]
```diff
@@ -371,6 +371,9 @@ void iso_filesystem::set_path_separator(char cSeparator) {
 
 bool iso_filesystem::initialise(const char* path) {
   clear();
+// Suppress invalid MSVC warning
+// https://github.com/CorsixTH/CorsixTH/issues/1961#issuecomment-868573582
+#pragma warning(suppress : 4996)
```
Comment (TheCycoONE) [inline CorsixTH/Src/iso_fs.cpp:376]: This applies to the whole file below this point. You may want to move it to the top of the file or undo the suppress after the line. If moving to the top of the file, link to the warning please.

Comment (TheCycoONE) [reply CorsixTH/Src/iso_fs.cpp:376]: Seems like the warning is a general security one so best suppressed just for this line. Also we shouldn't use this pragma outside of MSVC.

**Comment by TheCycoONE** [comment]
The MSVC thing didn't work:
```
C:\projects\corsixth\CorsixTH\Src\iso_fs.cpp(379,18): warning C4996: 'fopen': This function or variable may be unsafe. Consider using fopen_s instead. To disable deprecation, use _CRT_SECURE_NO_WARNINGS. See online help for details. [C:\projects\corsixth\CorsixTH\CorsixTH_lib.vcxproj]
```

You probably have to use the push/pop syntax because of the `#endif`

That said I'm fine if you want to leave that out and just handle the doxygen warnings right now.

**Comment by TheCycoONE** [comment]
Also perhaps consider _CRT_SECURE_NO_WARNINGS on the appveyor or in cmake instead.

Along with the other MSVC definition in CorixTH/CMakeLists.txt
```
add_definitions(-D_CRT_SECURE_NO_WARNINGS)
```

**Comment by tobylane** [comment]
The three doxygen commits could be merged, they're different levels of caution. It adds a few more warnings but I imagine the extra docs could be easily made by a C++ writer. Are any of them false positives? https://ci.appveyor.com/project/TheCycoONE/corsixth/builds/39766562#L258

**Comment by TheCycoONE** [comment]
LGTM

**Comment by tobylane** [comment]
Ready.

**Pull request merged**